protoc-gen-gogoroach/main.go

```diff
@@ -0,0 +1,41 @@
+// This is a customized version of https://github.com/gogo/protobuf/blob/master/protoc-gen-gogoslick/main.go
+
+package main
+
+import (
+	"github.com/gogo/protobuf/vanity"
+	"github.com/gogo/protobuf/vanity/command"
+)
+
+func main() {
+	req := command.Read()
+	files := req.GetProtoFile()
+
+	vanity.ForEachFile(files, vanity.TurnOnMarshalerAll)
+	vanity.ForEachFile(files, vanity.TurnOnSizerAll)
+	vanity.ForEachFile(files, vanity.TurnOnUnmarshalerAll)
+
+	// either one of these breaks compilation of the descriptor proto :(
+	// vanity.ForEachFieldInFiles(files, vanity.TurnOffNullableForNativeTypesOnly)
+	// vanity.ForEachFile(files, vanity.TurnOffGoGettersAll)
+
+	// do we want to support unknown fields?
+	vanity.ForEachFile(files, vanity.TurnOffGoUnrecognizedAll)
+
+	// prefixes usually make sense
+	// vanity.ForEachFile(files, vanity.TurnOffGoEnumPrefixAll)
+
+	vanity.ForEachFile(files, vanity.TurnOffGoEnumStringerAll)
+	vanity.ForEachFile(files, vanity.TurnOnEnumStringerAll)
+
+	// generates broken code
+	// vanity.ForEachFile(files, vanity.TurnOnEqualAll)
+
+	// we have some custom stringers
```

Comment:
Basically I think he has a good point.

Basically file level options overwrite vanity options, just like message level options overwrite file level options.
You could leave those marshaler, sizer extensions in.  The vanity binary will only change those if they don't exist.
So you could use a vanity binary just to make sure you don't forget extensions.

We are personally not using the vanity binaries yet.
I think vanity binaries are extremely cool for teams that want to have some conventions and for newbies.

I would really like to get my own team on pure vanity binaries, no importing of gogoproto, but that would mean changing all our protos to conform to some conventions, like a field named Id and type bytes is a customtype uuid.Uuid and all fields are nullable=false.  But what about those one or two fields that you actually want nullable=true.  And then you have to ask yourself don't you think nullable=false is quite explicit in a nice way and is all the effort really worth it if you can't completely remove the import of gogoproto and the descriptor.

You guys have some really nice use cases for gogoproto.  I am very impressed.  It is also really nice to your product being used in all its glory :)